### PR TITLE
Fix a bug with refreshing the gutter UI.

### PR DESCRIPTION
**Pull request by natthu**
I work on a package that adds a gutter to show "dot" next to a line (setting breakpoints). I noticed that sometimes the gutter UI would not update even if the code to decorate the marker successfully executed. When I debugged Atom, I realized that since the order in which gutters are added is undeterministic (unless you set the right priorities), `updateCustomGutterDecorationState` would sometimes return early.

### CODE REVIEW

**Comment by 50Wliu** [comment]
Can you please add specs for this?

**Comment by natthu** [comment]
Added a spec as requested.

File: spec/text-editor-presenter-spec.coffee [hunk 1]
```diff
@@ -2300,6 +2300,15 @@ describe "TextEditorPresenter", ->
 
             expect(decorationState[decoration3.id]).toBeUndefined()
 
+          it "updates when another gutter with higher priority is hidden", ->
```
Comment (as-cii) [inline spec/text-editor-presenter-spec.coffee]: :bulb: What about something along the lines of:

```coffee
it "updates all the gutters, even when a gutter with higher priority is hidden", ->
```

File: spec/text-editor-presenter-spec.coffee [hunk 2]
```diff
@@ -2300,6 +2300,15 @@ describe "TextEditorPresenter", ->
...
+            editor.addGutter({name: 'test-gutter-1', priority: -150, visible: false})
```
Comment (as-cii) [inline spec/text-editor-presenter-spec.coffee]: What about extracting a variable here, to make clear we are adding an hidden gutter? e.g.

```coffee
hiddenGutter = {name: 'any-name', priority: -150, visible: false}
editor.addGutter(hiddenGutter)
```

File: spec/text-editor-presenter-spec.coffee [hunk 3]
```diff
@@ -2300,6 +2300,15 @@ describe "TextEditorPresenter", ->
...
+
+            # This update will scroll decoration1 out of view, and decoration3 into view.
+            expectStateUpdate presenter, -> presenter.setScrollTop(scrollTop + lineHeight * 5)
+
+            decorationState = getContentForGutterWithName(presenter, 'test-gutter')
+            expect(decorationState[decoration1.id]).toBeUndefined()
```
Comment (as-cii) [inline spec/text-editor-presenter-spec.coffee]: What do you think about testing also for `decoration3` to be defined? :thought_balloon:

**Comment by as-cii** [comment]
I left a couple of comments, but otherwise it looks good. :ok_hand: 

Thanks @natthu for addressing this! :sparkles:

**Comment by as-cii** [comment]
Thank you, @natthu! Great job here! :fireworks:

**Pull request merged**